### PR TITLE
first pass at tfvars for simpler code base

### PR DESCRIPTION
**Pull request by smark88**
- switched to tfvars
- terrform 0.13.5 upgrade
- google provider update
- actions update
- updated readme.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,15 +6,18 @@ Terraform code to setup various GCP environments for the Jade Data Repo.
...
 ```sh
 git clone https://github.com/broadinstitute/terraform-jade.git
-cd terraform-jade/environments/<env>
-./mkEnv.sh -e <env> -s <suffix>
-./terraform.sh init
-./terraform.sh plan
-#./terraform.sh apply
+cd datarepo
+docker run --rm -it -v "$PWD":/working -v ${HOME}/.vault-token:/root/.vault-token broadinstitute/dsde-toolbox:consul-0.20.0 ./mkEnv.sh -e <env>
+./terraform.sh init -backend-config=bucket=<googe_project>
```
Comment (fboulnois) [inline README.md:11]:
```suggestion
./terraform.sh init -backend-config=bucket=<google_project>
```

File: README.md [hunk 2]
```diff
@@ -6,15 +6,18 @@ Terraform code to setup various GCP environments for the Jade Data Repo.
...
+./terraform.sh plan -var-file=tfvars/<env>.tfvars
+./terraform.sh apply -var-file=tfvars/<env>.tfvars
 ```
+## Variables
+- `<env>` should correspond to the environment to deploy (typically either `dev`
+or `prod`)
+- `<googe_project>` is for the statefile its the name of the google project typically
```
Comment (fboulnois) [inline README.md:18]:
```suggestion
- `<google_project>` is for the statefile its the name of the google project typically
```

File: .github/workflows/terraformPrPlan.yml [hunk 3]
```diff
@@ -37,7 +34,7 @@ jobs:
             --data '{"role_id":"'"${ROLE_ID}"'","secret_id":"'"${SECRET_ID}"'"}' \
             ${VAULT_ADDR}/v1/auth/approle/login | jq -r .auth.client_token)
           echo ::add-mask::${VAULT_TOKEN}
-          echo ::set-env name=VAULT_TOKEN::${VAULT_TOKEN}
+          echo "VAULT_TOKEN=$(echo ${VAULT_TOKEN})" >> $GITHUB_ENV
```
Comment (fboulnois) [inline .github/workflows/terraformPrPlan.yml:37]: If I understand correctly, you're pushing everything into an environment file called `${GITHUB_ENV}`? Also, any concerns about leaking credentials using `echo`? If you set a mask on `${VAULT_TOKEN}` I think you can simplify this to `echo "VAULT_TOKEN=${VAULT_TOKEN}"`

**Review by fboulnois — CHANGES_REQUESTED** [review]
I like this new setup, it's a lot simpler. A few suggestions:

File: .github/workflows/terraformPrPlan.yml (hunk 3 above)
Comment (smark88) [reply .github/workflows/terraformPrPlan.yml:37]: This is the new way to set env vars.

Comment (smark88) [reply .github/workflows/terraformPrPlan.yml:37]: also this did not work `echo "VAULT_TOKEN=${VAULT_TOKEN}" >> ${GITHUB_ENV}`

File: .github/workflows/terraformPrPlan.yml [hunk 4]
```diff
@@ -18,16 +19,12 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: "${{ matrix.environment }} Checkout"
-        uses: actions/checkout@master
+        uses: actions/checkout@v2
 
-      - name: "${{ matrix.environment }} Terraform Format"
-        uses: hashicorp/terraform-github-actions@v0.8.0
+      - name: install terrform
```
Comment (rhiananthony) [inline .github/workflows/terraformPrPlan.yml:24]: nit: typo

File: datarepo/terraform.sh [hunk 5]
```diff
@@ -1,8 +1,8 @@
 #!/bin/bash
 
-DEFAULT_TERRAFORM_VERSION="0.12.29"
+DEFAULT_TERRAFORM_VERSION="0.13.5"
 #DOCKER_IMAGE='broadinstitute/terraform:latest'
```
Comment (rhiananthony) [inline datarepo/terraform.sh:4]: Should you just remove the comment? Seems like it isn't necessary/ might be confusing.

File: datarepo/variables.tf [hunk 6]
```diff
@@ -2,23 +2,23 @@
...
 
 variable "k8_subnet_name" {
   description = "name of the subnet within the networking being deployed"
-  default     = "production-subnet"
+  default     = ""
 }
 
 variable "node_count" {
   type        = number
   description = "number of kubernetes nodes depends on if region is us-central1 1 will deploy 3 and us-central1-a 3 will deploy 3"
```
Comment (rhiananthony) [inline datarepo/variables.tf:20]: This description is confusing.

**Review by rhiananthony — CHANGES_REQUESTED** [review]

File: datarepo/terraform.sh (hunk 5 above)
Comment (smark88) [reply datarepo/terraform.sh:4]: yeah thats carry over form forever ago

**Comment by smark88** [comment]
Could use some thumbs. If it all looks good.

**Review by rhiananthony — APPROVED** [review]
LGTM

**Pull request merged**